source/include/ble/cs_BleCentral.h

```diff
@@ -0,0 +1,178 @@
+/*
+ * Author: Crownstone Team
+ * Copyright: Crownstone (https://crownstone.rocks)
+ * Date: Mar 26, 2021
+ * License: LGPLv3+, Apache License 2.0, and/or MIT (triple-licensed)
+ */
+
+#pragma once
+
+#include <ble/cs_Nordic.h>
+#include <ble/cs_UUID.h>
+#include <common/cs_Types.h>
+#include <events/cs_Event.h>
+#include <structs/cs_PacketsInternal.h>
+#include <events/cs_EventListener.h>
+
+/**
+ * Class that enables you to connect to a device, and perform write or read operations.
+ *
+ * Generally, you first connect, then discover, then perform read and/or write operations, and finally disconnect.
+ * Note that the disconnect event might happen at any time.
+ */
+class BleCentral: EventListener {
+public:
+	static BleCentral& getInstance() {
+		static BleCentral instance;
+		return instance;
+	}
+
+	void init();
+
+	/**
+	 * Connect to a device.
+	 *
+	 * @return ERR_BUSY                When already connected or another operation is in progress (discovery, read, write, connect, disconnect).
+	 * @return ERR_WAIT_FOR_SUCCESS    When the connection will be attempted. Wait for EVT_BLE_CENTRAL_CONNECT_RESULT.
+	 */
+	cs_ret_code_t connect(const device_address_t& address, uint16_t timeoutMs = 3000);
```

Comment:
Considering usability for crownstone-crownstone connections. Do we have a utiltiy to construct a `device_address_t` object for another crownstone given its `stone_id_t`? Perhaps a utility function overload would be nice?

```
cs_ret_code_t connect(const stone_id_t& address, uint16_t timeoutMs = 3000)
```

Reply:
Good point, should be implemented in CrownstoneCentral.